### DISCUSSION TITLE
Best practice for executing tests on windows environment

### DISCUSSION THREAD

**Discussion by jonathangiber** [getgauge/gauge#2250] [General]
Hi guys,
Im trying to run tests in a windows instance with gauge + Taiko (on google cloud / AWS).

The execution of taiko/gauge is happening on a unix instance on the same network of the windows instance (since we want to run in both unix and windows).

Should I use remote debugging on the windows machine with options.host and options.port or is there any other best practice for it?

Thanks

**Comment by jonathangiber** [comment]
Also, would like to understand if parallel execution will work in this case

**Comment by sriv** [comment]
If you have the dependencies installed, the windows execution should be the same as unix ones. Is there any difference in behaviour?

**Comment by jonathangiber** [comment]
Hi @sriv, thanks for answering.
Attached is an example on how we are running the tests in windows, since we got a requirement to run also in windows itself.

So basically I execute a docker with taiko on a linux machine that connect with the remote host (windows, that u can see on the attached image above)
![image](https://user-images.githubusercontent.com/7656511/173400351-eae04e93-1b7d-4212-9daa-200f5195b359.png)

I couldn't use the parallel execution of gauge while running in this way, thats why I was asking is there any other way to execute tests in windows, in unix I just run them from docker.

Thanks

**Reply by zabil** [reply]
This depends on how you've setup up Taiko's `openBrowser`. Because of how Taiko handles the browser instance it's not that easy to run tests in parallel. What issues do you exactly see?

**Reply by jonathangiber** [reply]
Hi @zabil,

This is how I defined openBrowser:

```
await openBrowser({
            headless: false,
            args: ["--proxy-server=http://" + getenv("TENANT_DOMAIN_NAME") + ":8080"],
            host: 'someIp',
            port: 9000,
            ignoreCertificateErrors: true
        });
```

Gauge tries to run in different runners but what actually happens in the windows remote host, is that all tests run in the same browser that is already open. Not more browsers are opened to run in parallel.

**Reply by zabil** [reply]
That usually happens when `openBrowser` is the `beforeSuite` hook. If that's the case use `openBrowser` before each scenario. There's an example over here https://github.com/getgauge-examples/js-taiko. Please note that the example uses a local instance. 

Also note, the sample above points to one port
```
port: 9000
```

Which means that it won't launch separate instances. You might have to figure out a way to assign the different browser ports.

**Comment by jonathangiber** [comment]
Thanks @zabil, I think we will try to execute directly in a windows instance without remote debugging or try with docker on a windows base image. This approach has too many complications.
